### ISSUE TITLE
[Bug]: ServiceWorker crash after restart

### ISSUE THREAD

**Issue by YuriL180821** [electron/electron#35899]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

17.0.0

### What operating system are you using?

Windows

### Operating System Version

Windows 10

### What arch are you using?

x64

### Last Known Working Electron version

16.0.0

### Expected Behavior

ServiceWorkerSingleScriptUpdateChecker::OnReceiveResponse should not crash when call 
```
base::MakeRefCounted<PolicyContainerHost>(
          // This does not parse the referrer policy, which will be
          // updated in ServiceWorkerGlobalScope::Initialize
          PolicyContainerPolicies(script_url_, response_head.get(), nullptr));
```
          

### Actual Behavior

After restart we have crash in 
policy_container_host_ = base::MakeRefCounted<PolicyContainerHost>(...)

### Testcase Gist URL

JS sample for testing can be taken from next link https://github.com/electron/electron/pull/34671#issuecomment-1162385596

### Additional Information

Problem is caused by absence of ::network::mojom::ParsedHeadersPtr into network::mojom::URLResponseHeadPtr after restart with the same local service worker script. 

In fact we have problem into that commit 
f6cd769a880007d524037a2d95736081a34ce704
Jonathan Hao <phao@chromium.org>
26.08.2022 12:18:21

It does not have check for empty ParsedHeadersPtr  into response

**Comment by SatwikKini** [comment]
I am a new contributor and trying to fix some bugs, I can please try to fix this one?

**Comment by YuriL180821** [comment]
I have solution already, but hope somebody with expertise will take a look on problem. It should be direct solution, I've tried to show mistakes in concept

**Comment by codebytere** [comment]
@YuriL180821 your explanation isnt quite clear - are you saying that https://chromium-review.googlesource.com/q/3766681 introduced this bug? if yes, has it been reported upstream?

**Comment by YuriL180821** [comment]
@codebytere  think my explanations above were enough clear, mentioned commit f6cd769a880007d524037a2d95736081a34ce704 introduces a bug which leads to electron crash in case of working with local service worker, even take into according that author mentioned that handled everything
Do You need realtime sharing where and what crashes?

**Comment by YuriL180821** [comment]
@codebytere should I expect any updates?

**Comment by clavin** [comment]
Hi, just taking a second look here. 👋🙂 Is this not a Chromium bug? That is, wouldn't this issue also happen in Chromium and the fix would need to be made there?

**Comment by YuriL180821** [comment]
Hello Community!
Hope this time my explanatiuons will be enough to understand the problem:
In his commit author solved this issue https://chromium-review.googlesource.com/c/chromium/src/+/3766681 i.e 
**Policy container inheritance for service workers**
so in my vision we did not had inheritance and after all changes we received it.
Also into his commit description author wrote next

```
- for previously loaded script which is stored in the database, we write
  the policy container policies to db just like the existing service
  worker data, hence the changes around service worker database and
  protocol buffer.
- I added a test to make sure that the encoding and decoding works

```
In fact I'm talking maybe about side effect or something not covered by author which causes next https://github.com/electron/electron/pull/35810#issuecomment-1275930347
Think crash into runtime is enough big problem to be qualified as blocker for release but not a bug.

**Comment by YuriL180821** [comment]
Fixed into 33f6c5e8272998d61c3337bc937054d37c54be3f
